File: problems/easy/1-two-sum/1-two-sum.cpp

/*
LeetCode 1. Two Sum
Difficulty: easy
URL: https://leetcode.com/problems/two-sum/

Решение добавлено: 2025-08-20
*/

#include <iostream>
#include <vector>
using namespace std;


vector<int> twoSum(vector<int>& nums, int target) {
    if (nums.empty()) {
        cout << "Vector is empty!" << endl;
        return {}; 
    }
    for(int i = 0; i < nums.size(); i++){
        for (int j = i+1; j < nums.size(); j++){
            if(nums[i] + nums[j] == target){
                return {i, j};
            }
        }
    }
    return {0, 0};
}

int main(){
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    cout << "Result: ";
    //twoSum(nums, target);
    vector<int> result = twoSum(nums, target);

    for (int i = 0; i < result.size(); i++){
        cout << result[i] << ", " ;
    }
    cout << endl;
    
}

File: problems/easy/1-two-sum/1-two-sum.py

"""
LeetCode 1. Two Sum
Difficulty: easy
URL: https://leetcode.com/problems/two-sum/

Решение добавлено: 2025-08-20
"""

def func(nums, target):
    for i in range(len(nums)):
        for j in range(i+1, len(nums)) :
            print(f'num1 = {nums[i]}, num2 = {nums[j]}')
            if nums[i] + nums[j] == target:
                return [i, j]
    return []

nums = [3, 2, 4]
target = 6
print(func(nums, target))

